src/sycltest/plugin-Test1/gpuAlgo1.cc

```diff
@@ -0,0 +1,175 @@
+#include <algorithm>
+#include <cmath>
+
+#include <CL/sycl.hpp>
+
+#include "SYCLCore/device_unique_ptr.h"
+#include "SYCLCore/host_unique_ptr.h"
+#include "gpuAlgo1.h"
+
+namespace gpu_algo_1 {
+  constexpr int NUM_VALUES = 1000;
+
+  template <typename T>
+  SYCL_EXTERNAL void vectorAdd(const T *a, const T *b, T *c, int numElements, sycl::nd_item<3> item_ct1) {
+    int i = item_ct1.get_group(2) * item_ct1.get_local_range().get(2) + item_ct1.get_local_id(2);
```

Comment:
If I understand correctly, the index `2` refers to the third element of `item_ct1`. I find using that (instead of `0` and first element) a bit counter-intuitive (but ok). Is that perhaps the closest correspondence with CUDA's x-dimension?

Reply:
Unfortunately yes, I think the order of the indices in SYCL is reversed with respect that in CUDA.

In CUDA, `x` increments faster, followed by `y`, followed by `z`.
In SYCL, `<2>` supposedly increments faster, followed by `<1>`, followed by `<0>`.
I write "supposedly" because I haven't read much about it (I haven't searched much, either), I only remember stumbling on a Codeplay [blog entry](https://www.codeplay.com/portal/blogs/2019/11/18/computecpp-v1-1-6-changes-to-work-item-mapping-optimization.html) related to this.

Anyway, the mapping was done automatically by the conversion tool, so I assumed it to be correct :-)

Reply:
I found some more details in the [SYCL documentation(https://www.khronos.org/registry/SYCL/specs/sycl-2020-provisional.pdf#equation.4.7.3):
![image](https://user-images.githubusercontent.com/4069793/95338262-0cb09780-08b3-11eb-9ae0-9d0830813e82.png)

Reply:
Thanks. The blog post explicitly mentions a transpose (even if not explicitly relative to CUDA).

Should we look into the SYCL-provided "linearization functions" (beyond this PR)?

Reply:
> Should we look into the SYCL-provided "linearization functions" (beyond this PR)?

Ah, I see now that you did already :)